pandas/core/series.py

```diff
@@ -1075,6 +1075,87 @@ def _set_value(self, label, value, takeable=False):
         return self
     _set_value.__doc__ = set_value.__doc__
 
+    def set_index(self, arrays, append=False, inplace=False,
```

Comment:
don't change the argument name, it is called keys currently, leave it that way

Reply:
The argument currently does not exist - it makes sense for `DataFrame` to have it names `keys`, since the main application (arguably) is setting the index to a given column *key*, but it would be very confusing for `Series`.

The signatures differ already anyway (no `drop` for `Series`), so I'm strongly -1 on naming this parameter `keys`.

Reply:
that doesn't make any sense, you are conflating 2 different things. change then name to match.

Reply:
That, to me, would be an argument to *not* share the implementation in `generic.py`. A parameter named `keys` makes zero sense for series (and neither does `drop` for that matter - or rather it would be very confusing).